docs/tutorial/building-a-theme.md

```diff
@@ -0,0 +1,1504 @@
+
+However, we don't have any event data in the site. Copy the `events.yml` file from `gatsby-theme-events/data` into `site/events`. Then, restart the dev server:
+
+![Running the site, the basepath is now `/events`](./images/building-a-theme-changed-basepath.png)
+
+The events listing page will now be accessible at [http://localhost:8000/events](http://localhost:8000/events). Hitting [the root path](http://localhost:8000/) will now return a 404.
+
+## Make themes extendable with gatsby-plugin-theme-ui
+
+You can make your theme styles extendable using the `gatsby-plugin-theme-ui` package.
+
+Install dependencies:
+
+```shell
+yarn workspace gatsby-theme-events add gatsby-plugin-theme-ui theme-ui @emotion/core @emotion/styled @mdx-js/react
+```
+
+Then, add the `gatsby-plugin-theme-ui` plugin to the `gatsby-theme-events/gatsby-config.js` file:
+
+```javascript:title=gatsby-theme-events/gatsby-config.js
+module.exports = ({ contentPath = "data", basePath = "/" }) => ({
+  plugins: [
+    // highlight-next-line
+    "gatsby-plugin-theme-ui",
+    {
+      resolve: "gatsby-source-filesystem",
+      options: {
+        path: contentPath,
+      },
+    },
+    {
+      resolve: "gatsby-transformer-yaml",
+      options: {
+        typeName: "Event",
+      },
+    },
+  ],
+})
+```
+
+`gatsby-plugin-theme-ui` takes a global theme context object and makes it available to all themes using `gatsby-plugin-theme-ui`.
+
+To use it, create a `theme.js` file in `gatsby-theme-events/src`:
+
+```javascript:title=gatsby-theme-events/src/theme.js
+export const theme = {
+  space: [0, 4, 8, 16, 32],
+  fonts: {
+    body: "-apple-system, BlinkMacSystemFont, Segoe UI, Roboto, sans-serif",
+  },
+  fontSizes: [16, 18, 20, 22, 27, 36],
+  lineHeights: {
+    body: 1.45,
+    heading: 1.1,
+  },
+  colors: {
+    gray: ["#efefef", "#ddd", "#333", "#111"],
+    background: "#fff",
+    primary: "rebeccapurple",
+  },
+  sizes: {
+    default: "90vw",
+    max: "540px",
+  },
+  styles: {
+    Layout: {
+      color: "gray.2",
+      fontFamily: "body",
+      fontSize: 1,
+      lineHeight: "body",
+    },
+    Header: {
+      backgroundColor: "primary",
+      color: "background",
+      fontWeight: "bold",
+      margin: "0 auto",
+      maxWidth: "max",
+      padding: 3,
+      width: "default",
+      a: {
+        color: "inherit",
+      },
+    },
+    Main: {
+      margin: "0 auto",
+      maxWidth: "max",
+      width: "default",
+    },
+    Container: {
+      padding: 3,
+    },
+    h1: {
+      color: "gray.3",
+      fontSize: 5,
+      fontWeight: "bold",
+      lineHeight: "heading",
+      margin: "1rem 0 0",
+    },
+    ul: {
+      borderTop: "1px solid",
+      borderColor: "gray.0",
+      listStyle: "none",
+      padding: 0,
+    },
+    li: {
+      borderBottom: "1px solid",
+      borderColor: "gray.1",
+      padding: 2,
+      "&:focus-within,&:hover": {
```

Comment:
We should add a note about polyfilling `:focus-within` here or use something that's better supported.

Reply:
Open to suggestions! This is copied from the Egghead course.

Reply:
I should have just dropped this entirely — it's only there to show how to style things, but this is too much style to bundle into a basic theme.